Feature/lumina node integration 1

gabros20 opened the pull request.

jcstein commented:
Lgtm! Could you remove the second former in my blurb please?

jcstein commented on src/datas/team/team-members.js:
```diff
@@ -339,5 +312,31 @@ export const teamMembers = [
+        id : 28,
+        name : 'Josh Stein',
+        title: 'Developer Relations',
+        text: 'Former DevRel at Ankr & former Teaching Assistant at buildspace',
```
```suggestion
        text: 'Former DevRel at Ankr & Teaching Assistant at buildspace',
```

jcstein commented on src/datas/team/team-members.js:
```diff
@@ -339,5 +312,31 @@ export const teamMembers = [
+        text: 'Former DevRel at Ankr & former Teaching Assistant at buildspace',
+        image: 'team/JoshStein.png',
+        socials: {
+            twitter : '',
```
```suggestion
            twitter : 'https://twitter.com/JoshCStein',
```

jcstein reviewed (changes requested):
added some comments, lgtm!

gabros20 committed:
feat: Add NodeStatus component and related SVG icons

- Created NodeStatus component for displaying block verification status
- Added LuminaCheckmarkSVG, LuminaDiagonalArrowSVG, LuminaErrorSVG, and LuminaGradientCircleSVG
- Integrated NodeStatus into DesktopNav
- Updated .gitignore to exclude docs/ directory

gabros20 committed:
chore: Update Next.js config and SVG components

- Add Next.js configuration with page extensions, image optimization, and redirects
- Fix SVG components by updating SVG attribute syntax (stroke-* to strokeLinecap/strokeLinejoin)
- Migrate inline styles to React object syntax in LuminaGradientCircleSVG
- Sanitize JSON-like attributes in SVG components

gabros20 committed:
feat: working connection

gabros20 committed:
fix: loader svg animation direction

gabros20 committed:
fix: updated loader svg

gabros20 committed:
fix: multiple hydration issues in the original project build

gabros20 committed:
feat: Enhance NodeStatus component with dynamic syncing progress and improved status messaging

gabros20 committed:
chore: Refactor blog fetching, NodeStatus, and navigation components

- Improve blog post fetching with error handling and caching
- Update NodeStatus component styling and progress display
- Relocate AutoConnectingLuminaNode component
- Simplify LuminaDiagonalArrowSVG icon

gabros20 committed:
feat: Enhance animation and navigation components

- Refactor MovingGradient component to generate random gradient animations on client-side
- Update AutoConnectingLuminaNode to manage initialization state
- Modify NodeStatus to include animation controls and improved syncing messages
- Introduce MenuButton component for mobile navigation with SVG icons
- Clean up Nav component to integrate MenuButton and streamline layout

gabros20 committed:
fix: wasm issues build time

gabros20 committed:
feat: Update NodeStatus and useLuminaNode for improved syncing display

- Refactor NodeStatus to utilize syncProgress for approximate syncing percentage display as much possible
- Remove fake progress logic and simplify state management
- Enhance useLuminaNode hook to calculate and provide sync progress and connected peers
- Update event handling for better status updates during syncing and connection

gabros20 committed:
Merge branch 'main' into feature/lumina-node-integration

gabros20 committed:
Merge branch 'main' into feature/lumina-node-integration

gabros20 committed:
Update package-lock.json and EcosystemExplorer.js: bump package versions and clean up imports

gabros20 committed:
fix: connection issues

gabros20 committed:
fix: inifinie loop

gabros20 committed:
fix loading state indication

gabros20 committed:
feat: working block loading

gabros20 committed:
feat: new block rendering component added

gabros20 committed:
Merge branch 'feature/lumina-node-v2' into feature/lumina-node-integration

gabros20 committed:
commdent

gabros20 committed:
fix: live block render

gabros20 committed:
fix: build error

gabros20 committed:
Merge branch 'feature/test-lumina-node' into feature/lumina-node-integration

gabros20 committed:
Merge branch 'main' into feature/lumina-node-integration

gabros20 committed:
fix: animations

gabros20 committed:
feat: add sync percentage display and improve block number handling

gabros20 committed:
fix: block number visibility issues on mobile

gabros20 committed:
fix: mobile element sizes

gabros20 committed:
feat: enhance AutoLuminaContext with start/stop functionality and improve BlockNumberDisplay with control buttons
- Added startNode and stopNode methods to AutoLuminaContext for managing node synchronization.
- Updated BlockNumberDisplay to include start and stop buttons, improving user interaction.
- Adjusted useLuminaNode hook to handle new start/stop logic and updated status management.
- Enhanced UI responsiveness for mobile and desktop views.

gabros20 committed:
feat: make start/stop buttons visible on mobile and optimize banner widths - Remove hidden class from start/stop buttons to show on mobile - Add responsive button sizing: 28px mobile, 36px desktop - Add responsive icon scaling: 75% mobile, 100% desktop - Reduce idle state widths: 135px mobile, 204px desktop - Update mobile width calculations for button visibility - Adjust margins for smaller mobile button sizes

gabros20 committed:
fix: prevent stop button from showing during initializing phase - Change stop button logic to only show during syncing with actual data - Require blockNumber or syncPercentage > 0 before showing stop button - Prevent stop button during 'Initializing connection' phase

gabros20 committed:
fix: improve status text animation timing when stopping sync - Add longer delay (0.3s) for idle state text to prevent overlap with stop button exit - Fix animation direction: text now animates from top (-3px) instead of bottom (3px) - Add proper exit animations for smoother transitions - Prevent 'Start Light Node' text from jumping when stop button animates out

gabros20 committed:
feat: optimize banner width and make start/stop buttons visible on mobile - Reduce idle state width: 135px mobile, 204px desktop - Make start/stop buttons visible on mobile with responsive sizing - Add responsive button sizing: 28px mobile, 36px desktop - Add responsive icon scaling: 75% mobile, 100% desktop - Update margin calculations for smaller mobile buttons

gabros20 committed:
fix: Lumina banner width and start button positioning

- Updated banner width calculations for better button positioning

- Fixed start button spacing and alignment in idle state

- Improved margin calculations for all button states (start/stop/explorer)

- Changed block number display from opacity to block/hidden for cleaner visibility

- Refined stop button to only show during syncing (not initializing)

- Preserved all existing sync sequence and block number display logic

gabros20 committed:
fix: enhance error handling and improve sync display logic

gabros20 committed:
feat: updated ux flow, animation, structural refactor, mobile fixes

gabros20 committed:
feat: modules update

gabros20 committed:
Merge branch 'main' into feature/lumina-node-integration-1

gabros20 committed:
fix: buidl errors

gabros20 committed:
feat: added debug tool to track events and data flow

gabros20 committed:
feat: fix mobile view layout

alex-beckett committed:
Update HomeClient.js

gabros20 committed:
fix: text width for older ios devices

gabros20 committed:
fix: banner text wrapping bug on older ios browser

gabros20 committed:
feat: add ios version based rendering for banner

gabros20 committed:
feat: update lumina node package to latest version

gabros20 committed:
fix: local storage permissions only  trigger when lumina node started

gabros20 committed:
chore: remove docs

gabros20 committed:
fix: remove warning message on mobile

gabros20 committed:
fix: build error

gabros20 committed:
Merge remote-tracking branch 'origin/main' into feature/lumina-node-integration-1

gabros20 committed:
fix: build error
